lib/maze/servlets/command_servlet.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module Maze
+  module Servlets
+
+    # Receives log requests sent from the test fixture
+    class CommandServlet < BaseServlet
+      # Serves the next command, if these is one.
+      #
+      # @param _request [HTTPRequest] The incoming GET request
+      # @param response [HTTPResponse] The response to return
+      def do_GET(_request, response)
+        $logger.info 'Asked for a command'
+
+        response.header['Access-Control-Allow-Origin'] = '*'
+
+        commands = Maze::Server.commands
+        if commands.empty?
+          response.body = 'No commands to provide'
+          response.status = 400
+        else
+          response.body = JSON.pretty_generate(commands.current)
```

Comment:
Does `.current` get the head or tail of the list?

I'm not planning to queue up multiple commands, but curious about the behaviour.

Reply:
It should work in the natural way - add 3 things to queue and GET requests to `/commands` will return each in the order that they we added.  It's exactly the same mechanism that's used when expecting more than one error/session, with the `I discard the oldest log` Cucumber step moving to the next in the list.